app/controllers/admin/users_controller.rb

```diff
@@ -37,16 +37,13 @@ def update
 
     def upload
       begin
-        data = process_file_upload
+        data = process_file_upload(['.csv'])
+        @current_job = UploadUsersJob.perform_later(EndUser,
```

Comment:
See my comment in the implementation of `process_file_upload`. After you do that, I think you'll be able to move this code back into the `else` branch that you originally had.